packages/dd-trace/src/heap_snapshots.js

```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+async function scheduleSnapshot (config, total) {
+  if (total > config.heapSnapshot.count) return
+
+  const { setTimeout } = require('timers/promises')
+  const { writeHeapSnapshot } = require('v8')
+
+  await setTimeout(config.heapSnapshot.interval * 1000, null, { ref: false })
+  await clearMemory()
+  writeHeapSnapshot(getName(config.heapSnapshot.folder))
+  await scheduleSnapshot(config, total + 1)
+}
+
+async function clearMemory () {
+  const { setImmediate } = require('timers/promises')
+
+  globalThis.gc()
+  await setImmediate()
+  globalThis.gc() // Run full GC a second time for anything missed in first GC.
+}
+
+function pad (value) {
+  return String(value).padStart(2, 0)
+}
+
+function getName (folder) {
+  const { join } = require('path')
+  const { format } = require('util')
+  const { threadId } = require('worker_threads')
+
+  const date = new Date()
+  const filename = format(
+    'Heap-%s%s%s-%s%s%s-%s-%s.heapsnapshot',
+    date.getFullYear(),
+    pad(date.getMonth()),
+    pad(date.getDate()),
+    pad(date.getHours()),
+    pad(date.getMinutes()),
+    pad(date.getSeconds()),
+    process.pid,
+    threadId
+  )
```

Comment:
Unfortunate that we have to replicate this from V8 just because we want to specify the folder. It would be nice if specifying a folder without having to specify the filename was supported by V8

Reply:
Hm... yeah, it would probably be a good idea to accept a folder in the Node.js API and just automatically write them there with the default name instead of having to pass through the full name.